### PR TITLE
Improve make_poller_files.py

### PR DESCRIPTION
**Pull request by dulude**
**Relevant Ticket**
-  [HLA-572](https://jira.stsci.edu/browse/HLA-572)

**Summary of changes**
This PR addresses issues raised by @mdlpstsci  regarding a couple of UX bugs in make_poller_files.py. Fixes/improvements include the following:
- Added additional logic to throw an exception if users don't specify a skycell name when creating a MVM poller file
- Added additional logic to throw an exception if users don't specify a properly formatted skycell name when creating a MVM poller file
- Refactored the fits file location subroutine to make it more flexible and the logic more robust.
- Assorted docstring improvements and updates

### CODE REVIEW

File: drizzlepac/haputils/make_poller_files.py [hunk 1]
```diff
@@ -117,35 +133,39 @@ def locate_fitsfile(search_string):
          full file path + image name of specified search_string.
     """
     if search_string.endswith("_flt.fits") or search_string.endswith("_flc.fits"):  # Process search_string as a full filename
-        # Look for files in CWD first
-        if os.path.exists(search_string):
+        # Look in user-provided path (assuming they provided one)
+        if os.path.exists(search_string) and "/" in search_string:
```
Comment (stsci-hack) [inline drizzlepac/haputils/make_poller_files.py:137]: The use of the hard-coded value '/' here should be replaced instead with 'os.sep' in order to make this code more platform-independent.

File: drizzlepac/haputils/make_poller_files.py [hunk 2]
```diff
@@ -117,35 +133,39 @@ def locate_fitsfile(search_string):
...
+            return search_string
+        # Look for files in CWD
+        if os.path.exists(search_string) and "/" not in search_string:
             return os.getcwd()+"/"+search_string
         # If not found in CWD, look elsewhere...
         if not os.getenv("DATA_PATH"):
             sys.exit("ERROR: Undefined online cache data root path. Please set environment variable 'DATA_PATH'")
-        fullfilepath = "{}/{}/{}/{}".format(os.getenv("DATA_PATH"), search_string[:4], search_string, search_string)
+        fullfilepath = "{}/{}/{}/{}".format(os.getenv("DATA_PATH"), search_string[:4], search_string[:-9], search_string)
```
Comment (stsci-hack) [inline drizzlepac/haputils/make_poller_files.py:145]: Again, a more platform-independent way to generate filenames would be to use 'os.sep'..

**Review by stsci-hack — CHANGES_REQUESTED** [review]
Please take a look at making this code work on all Python platforms as noted in my comments.

File: drizzlepac/haputils/make_poller_files.py (hunk 1 above)
Comment (dulude) [reply drizzlepac/haputils/make_poller_files.py]: issue resolved by commit [6626a45](https://github.com/spacetelescope/drizzlepac/pull/1138/commits/6626a452591a0f7331157f0954be141be06cfa4f)

File: drizzlepac/haputils/make_poller_files.py (hunk 2 above)
Comment (dulude) [reply drizzlepac/haputils/make_poller_files.py]: issue resolved by commit [6626a45](https://github.com/spacetelescope/drizzlepac/pull/1138/commits/6626a452591a0f7331157f0954be141be06cfa4f)

**Review by stsci-hack — APPROVED** [review]
LGTM

**Pull request merged**